### PR TITLE
Changes requested by Aileen Nielsen

### PR DESCRIPTION
**Pull request by vyasr**
I received a few other change requests via email. In the interests of time, I'll aim to merge this PR by EOD Wednesday.

### CODE REVIEW

**Comment by vyasr** [comment]
For reference, the file she commented on is attached here. The comments are listed as -- TODO or --TODO. Note that I renamed it as a TXT file so that I could upload, but it is a valid rst file.
[94_vyas_ramasubramani.txt](https://github.com/vyasr/scipy_proceedings/files/2135127/94_vyas_ramasubramani.txt)

File: papers/vyas_ramasubramani/94_vyas_ramasubramani.rst [hunk 1]
```diff
@@ -188,6 +191,8 @@ Additionally, at any point we can get an overview of what the implicit data spac
      'v': 'int([1, 2, 3], 3)',
     }
 
+Additionally, schema can be filtered, nested keys can be compressed to specified depths, and the number of entries shown in each range can be limited as desired.
```
Comment (csadorf) [inline papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: You should mention that all keys that have *constant* values across all data points can be automatically filtered so as to only see "relevant" schema data.

File: papers/vyas_ramasubramani/94_vyas_ramasubramani.rst [hunk 2]
```diff
@@ -328,8 +333,9 @@ For better performance, the resulting indexes can be manually saved; in general,
...
-In addition to the core index-related functionality previously mentioned, the ``signac`` ``Project`` also encapsulates numerous additional features, including, for example, the detection of implicit schema; the generation of human-readable views of the hash-obfuscated workspace; the ability to move, copy, or clone a full project; and the ability to synchronize data across projects.
-Individual data points, which can be accessed by searching through or iterating over a ``Project`` instance, are represented by ``Job`` objects.
+In addition to the core index-related functionality previously mentioned, the ``signac`` ``Project`` also encapsulates numerous additional features, including, for example, the generation of human-readable views of the hash-obfuscated workspace; the ability to move, copy, or clone a full project; the ability to synchronize data across projects; and the detection of implicit schema.
+Note that the schema are implicit insofar as they are only defined by the state points of jobs within the workspace, *i.e* there is nothing like a table schema to enforce a particular structure for the state points of individual jobs.
```
Comment (csadorf) [inline papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: I would rephrase:

> We qualify the schema as implicit, as ...

File: papers/vyas_ramasubramani/94_vyas_ramasubramani.rst [hunk 3]
```diff
@@ -191,6 +191,7 @@ At any point, we can also get an overview of what the implicit data space schema
      'v': 'int([1, 2, 3], 3)',
     }
 
+Keys with constant values across the entire data space are automatically omitted from the schema.
```
Comment (csadorf) [inline papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: *optionally*

Comment (vyasr) [reply papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: Whoops thanks. I'll fix.

File: papers/vyas_ramasubramani/94_vyas_ramasubramani.rst [hunk 4]
```diff
@@ -329,10 +329,13 @@ To ensure that the state point associated with the job id can be recovered, a JS
...
-For better performance, the resulting indexes can be manually saved; in general, ``signac`` automatically caches generated indexes for this purpose.
+In general, ``signac`` automatically caches generated indexes within a single session where possible, but for better performance the indexes can also be manually saved.
 These indexes then allow efficient selection and searching of the data space, and MongoDB-style queries can be used for complex selections.
 
+This distributed mode of operation is well-suited to the high performance filesystems common to high performance computing.
+The implicit partitioning of the data on a per-job basis ensures that any operations that do not require explicit indexing will be fast and nearly scale-invariant with respect to the data space size.
+At present, ``signac`` does not support explicit horizontal partitioning of the data space, but because a ``signac`` project can always be linked with a new workspace, manually partitioning the data space so as to emulate sharding is a relatively simple task that simply requires the user to determine how the data should be divided between shards.
```
Comment (csadorf) [inline papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: Since the state point represents the primary key and if we interpret the document as one row within the table, then the signac data layout is even for one workspace essentially horizontally partitioned. However, I think that partitioning the data manually is less relevant here, than the fact that the data is inherently distributed and may be accessed in parallel by potentially thousands of clients in parallel. This is why it suits typical HPC workflows very well.

How about this:
> The explicit horizontal partitioning and distributed storage of data per job is well suited to typical HPC operations, which are executed for multiple jobs in parallel.
> Since data is accessed distributively there is no inherent bottleneck posed by funneling all data read and write operations through one or more server applications.

Comment (vyasr) [reply papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: I agree that for our purposes the inherent distribution of the data is the important point, especially for HPC. However, the particular question this was meant to address was relating to sharding, which typically encompasses horizontal partitioning across filesystems. While one benefit is to reduce load, which is automatically addressed by signac as is, another is potentially to reduce the total index size, or perhaps to make better use of available space. In that context, it would be helpful to indicate how the data space can be actually partitioned. I like your phrasing, but I think it would be good to incorporate a sentence discussing that aspect as well, which is what my last sentence was meant to address. Do you think we can combine the two versions?

Comment (vyasr) [reply papers/vyas_ramasubramani/94_vyas_ramasubramani.rst]: @csadorf I've tried to use your version and add a sentence to more explicitly address the question. Take a look when you get a chance and we can finalize.

**Comment by csadorf** [comment]
I like the latest changes.
